### DISCUSSION TITLE
Blocking

### DISCUSSION THREAD

**Discussion by RngesusSolutions** [256shadesofgrey/eso-light-attack-weave#25] [General]
I have been attempting to edit this script to suspend when blocking as it causes issues with block casting many skills.

I have tried setting 

"global suspendKey := "^RButton"

But to no avail

**Comment by RngesusSolutions** [comment]
To add further context as I have changed the block canceling function to a key F so instead of block canceling it will bash cancel which is much more efficient now

**Comment by 256shadesofgrey** [comment]
> I have been attempting to edit this script to suspend when blocking as it causes issues with block casting many skills.

What exactly is the issue? Because if you set the correct block key [here](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L19), the script should not attempt to weave if you're holding down that key. When the skill comes up in the queue, it checks [here](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L238) if you're holding down the block button, and skips the weaving if you do. The skill still gets queued though, and if that is undesirable, you can add the same check [here](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L204).

> I have tried setting
> 
> "global suspendKey := "^RButton"
> 
> But to no avail

That is control+right mouse button key combination. I don't think that's what you wanted. You probably wanted just `global suspendKey := "RButton"`. But even then it's not going to do what you want, because it will disable when you press RButton once, and then re-enable when you press it a 2nd time instead of doing so when you release. The suspend keybind works as a toggle, not as a modifier. To prevent the script getting in the way of blocking I implemented the checks that I described above, and that would be the right place to look at.



> To add further context as I have changed the block canceling function to a key F so instead of block canceling it will bash cancel which is much more efficient now

The block cancelling is something I personally never use. I just implemented it, tested it, it seemed to work, disabled it, and never looked at it again. So it's possible that there are bugs that I missed. But the simplest way of implementing what you describe would be to add another variable (let's call it `bash`) to the list of variable definitions [here](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L18) and use that variable [here](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L252) and [here](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L254) instead of block. Or in fact change [this](https://github.com/256shadesofgrey/eso-light-attack-weave/blob/0b330dd8a72aea087908ce8b58d3191efd3c06bc/eso-light-attack-weave.ahk#L252) to `Send, {%bash%}` and delete the following 2 lines entirely. That would be the quick and hacky way that should work.


P.S. Since this isn't actually an issue with the script as I uploaded it, but an issue with your modification of the script, I will turn this into a discussion.
